lib/core/src/Cardano/Wallet/Primitive/Types/UTxOIndex/Internal.hs

```diff
@@ -373,18 +392,12 @@ disjoint i1 i2 = universe i1 `Map.disjoint` universe i2
 
 -- | Specifies a filter for selecting UTxO entries.
 --
-data SelectionFilter
-    = Any
-        -- ^ Select any UTxO entry from the entire set.
-    | WithAdaOnly
-        -- ^ Select any UTxO entry that only has ada and no other assets.
-    | WithAsset AssetId
-        -- ^ Select any UTxO entry that has a non-zero quantity of the specified
-        -- asset.
-    | WithAssetOnly AssetId
-        -- ^ Select any UTxO entry that has a non-zero quantity of the specified
-        -- asset, but no other non-ada assets.
-    deriving (Eq, Show)
+data SelectionFilter asset
+    = SelectSingleton asset
+    | SelectPairWith asset
+    | SelectAnyWith asset
+    | SelectAny
+    deriving (Eq, Foldable, Functor, Show, Traversable)
```

Comment:
Request: add some Haddock comments to each of the constructors in `SelectionFilter`.

Reply:
Fixed in 21a617aa480fdf363b473d0a4bf5dece2764e313.